Source/Plugins/Core/com.equella.core/js/tsrc/schema/SchemaModule.tsx

```diff
@@ -123,3 +125,43 @@ export const pathForNode = (node: SchemaNode, stripXml = true): string => {
 
   return path;
 };
+
+/**
+ * Recursive helper function to get a list of all possible xml paths from a given SchemaNode.
+ * @param nodes The schema to generate paths for.
+ * @param paths Used by the recursive algorithm to build up the returned list.
+ *          When passing in initially, leave as a blank array.
+ * @param stripXml Passed into pathForNode. Whether to include the leading /xml.
+ *
+ */
+export const getAllPaths = (
+  nodes: SchemaNode,
+  paths: string[],
+  stripXml = true
+) => {
+  paths.push(pathForNode(nodes, stripXml));
+  nodes.children?.forEach((childNode) => {
+    paths.concat(getAllPaths(childNode, paths, stripXml));
+  });
+
+  return paths;
```

Comment:
maybe leverage `flatMap` with `concat` to add

```suggestion
  return paths.concat(pathForNode(nodes, stripXml)).concat(
    nodes.children?.flatMap((childNode) => {
      return getAllPaths(childNode, paths, stripXml);
    }) ?? []
  );
```

Reply:
This will put an empty array into `paths` if `nodes.children` is undefined.

Reply:
It doesn't, line 142 uses `concat`.
when concat is passed an empty array it doesn't add anything.
Try it out in your browser console.

```js
['example'].concat([])
```

Reply:
Ah  I did a wrong experiment. I tried `['example'].concat(['a', []])`  and got `["example", "a", Array []]`.